pkg/istio/onboarding/controller.go

```diff
@@ -0,0 +1,246 @@
+package onboarding
+
+import (
+	"errors"
+	"log"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+
+	"istio.io/api/rbac/v1alpha1"
+	"istio.io/istio/pilot/pkg/model"
+)
+
+const (
+	numRetries             = 3
+	authzEnabled           = "true"
+	authzEnabledAnnotation = "authz.istio.io/enabled"
+)
+
+type Controller struct {
+	store                model.ConfigStoreCache
+	dnsSuffix            string
+	serviceIndexInformer cache.SharedIndexInformer
+	queue                workqueue.RateLimitingInterface
+}
+
+type metaNamespaceKeyFunc func(obj interface{}) (string, error)
+
+// processEvent processes the service watch events and puts them into the queue
+func (c *Controller) processEvent(fn metaNamespaceKeyFunc, obj interface{}) {
+	key, err := fn(obj)
+	if err == nil {
+		c.queue.Add(key)
+		return
+	}
+	log.Println("Error converting object to key:", err)
+}
+
+// NewController initializes the Controller object and its dependencies
+func NewController(store model.ConfigStoreCache, dnsSuffix string, serviceIndexInformer cache.SharedIndexInformer) *Controller {
+	queue := workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter())
+
+	c := &Controller{
+		store:                store,
+		dnsSuffix:            dnsSuffix,
+		serviceIndexInformer: serviceIndexInformer,
+		queue:                queue,
+	}
+
+	serviceIndexInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			c.processEvent(cache.MetaNamespaceKeyFunc, obj)
+		},
+		UpdateFunc: func(old interface{}, new interface{}) {
```

Comment:
New surprisingly isn't a keyword but new() is a function which is supported which initializes an object with empty values. I have renamed it to avoid confusion.